### PR TITLE
feat: promise.all

### PR DESCRIPTION
**Pull request by vlkerag**
## What does this change?
Monitoring:
- Wrap puppeteer into try catch finally to avoid memory leak in case of failure
- Clean up promise.all to not include an inner await or page.load/page.close

## Why?
- to avoid "Navigation failed because browser has disconnected!" or similar error.

### CODE REVIEW

**Comment by vlkerag** [comment]
working in code

File: cdk/lib/monitoring.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class Monitoring extends GuStack {
 		});
 
 		const monitoringDuration: Duration =
-			stage === 'PROD' ? Duration.minutes(2) : Duration.days(1); // Every day for CODE; Every 2 minutes for PROD.
+			stage === 'PROD' ? Duration.minutes(2) : Duration.minutes(2); // Every day for CODE; Every 2 minutes for PROD.
```
Comment (kenoir) [inline cdk/lib/monitoring.ts:74]: Presumably this is for testing? Does it need to be switched back to a lower duration before merge?

**Review by kenoir — DISMISSED** [review]
Looks good, one comment on the frequency in CODE that probably needs addressing.

File: monitoring/src/check-page/tcfv2.ts [hunk 2]
```diff
@@ -133,12 +137,15 @@ const checkPages = async (config: Config, url: string, nextUrl: string) => {
 	await checkCMPLoadingTime(page, config);
 
 	await page.close();
-
+} catch (e) {
```
Comment (kenoir) [inline monitoring/src/check-page/tcfv2.ts:140]: I think the formatting here is a little funky, I don't think this project has a "format" task to run prettier.

**Comment by kenoir** [comment]
Worth noting that some of the test logic here is duplicated (intentionally) in https://github.com/guardian/commercial-canaries/tree/main, should any of these changes also be in that repo?

**Comment by kenoir** [comment]
> ## What does this change?
> Wrap page test in promise.all
> 
> ## Why?
> to avoid "Navigation failed because browser has disconnected!"

It's a bit difficult to tell the what/why from outside the team without more context, a bit more information would be helpful.

File: cdk/lib/monitoring.ts (hunk 1 above)
Comment (vlkerag) [reply cdk/lib/monitoring.ts:74]: Yes, it is only for testing and will be reverted before prod

**Comment by vlkerag** [comment]
[Upgrade to node 18, upgrade to ts 5.1.3, upgrade as many dependencies as possible](https://trello.com/c/N042RtTX/1042-upgrade-to-node-18-upgrade-to-ts-513-upgrade-as-many-dependencies-as-possible)

**Comment by vlkerag** [comment]
[Upgrade to node 18, upgrade to ts 5.1.3, upgrade as many dependencies as possible](https://trello.com/c/N042RtTX/1042-upgrade-to-node-18-upgrade-to-ts-513-upgrade-as-many-dependencies-as-possible)

**Review by sookburt — APPROVED** [review]

**Pull request merged**